FR #1070: Clarify Commitment Discount concepts and glossary entries (#1964)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/glossary.md

<a name="glossary:commitment-discount"><b>Commitment Discount</b></a>

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one Commitment Discount may be applied to a single charge at a time, and it may overlap with one or more Negotiated Discounts.

Reviewer: Avoid capitalization of concepts.  (Apply in the other glossary entry too.)

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one Commitment Discount may be applied to a single charge at a time, and it may overlap with one or more Negotiated Discounts.

Author: ```suggestion
A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount).
```

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount).

Reviewer 2: ```suggestion
A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount). This term has been used in the past by FinOps teams managing Public Cloud to refer to specific contract commitments (including Reserved Instances, Savings Plans and Committed Use Discounts) offered by cloud providers, and is defined here to maintain consistency with the terms of that subset of contract commitments. 
```

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount). This term has been used in the past by FinOps teams managing Public Cloud to refer to specific contract commitments (including Reserved Instances, Savings Plans and Committed Use Discounts) offered by cloud providers, and is defined here to maintain consistency with the terms of that subset of contract commitments. 

Author: ```suggestion
A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount). 

This term has been used in the past by FinOps teams managing Public Cloud to refer to specific *contract commitments* (including Reserved Instances, Savings Plans and Committed Use Discounts) offered by cloud providers, and is defined here to maintain consistency with the terms of that subset of *contract commitments*. 
```

Commitment discounts are classified with the following designations:

* [Contract Commitment Offer Category](#datasets.contractcommitment.contractcommitmentoffercategory): "Public"
* [Contract Commitment Benefit Category](#datasets.contractcommitment.contractcommitmentbenefitcategory): "Discount"

> * [Contract Commitment Model](#datasets.contractcommitment.contractcommitmentmodel): "Continuous

Reviewer: ```suggestion
* [Contract Commitment Model](#datasets.contractcommitment.contractcommitmentmodel): "Continuous"
```

<a name="glossary:commitment-discount"><b>Commitment Discount</b></a>

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms that are generally available to all customers via the service provider's standard rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount).

Reviewer: These two definitions next to each other run very similarly until the second half.  Can we add a modifier that immediately differentiates them?  For example:

```suggestion
A publicly available [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), based on standard terms accessed via the service provider's rate card or portal. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount).
```

Author: I like the direction of this proposal and would suggest taking it one step further:

> **Commitment Discount:** A publicly available [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), based on publicly disclosed standard terms and pricing. ...
>
> **Negotiated Discount:** A negotiated [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified from standard. ...

### Reasoning

#### Commitment Discount:

* `based on standard terms accessed via the service provider's rate card or portal` -> `based on publicly disclosed standard terms and pricing`
  * removes reference to specific access channels (rate card, portal - implementation details); focuses on the key characteristic that terms are publicly available

#### Negotiated Discount:

* `privately agreed`" -> `negotiated`:
  * aligns directly with the allowed value `Negotiated` (not `Private`)
* `specifically modified from standard rates` -> `specifically modified from standard`
  * `rates` is dropped because it implies only pricing (rate card) has been modified, whereas a negotiated commitment may involve modifications to both pricing and broader contractual terms that go beyond the rate card

@shawnalpay, @robmartin33, thoughts?

```suggestion
A publicly available [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing on preselected SKUs in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), based on publicly disclosed standard terms and pricing. Committed spend or usage is evenly distributed across predefined [Contract Commitment Fulfillment Intervals](#datasets.contractcommitment.contractcommitmentfulfillmentinterval) (e.g., hourly), and unused benefits cannot be carried over to subsequent Intervals. Only one commitment discount may be applied to a single charge at a time, and it may overlap with one or more [*negotiated discounts*](#glossary:negotiated-discount).
```

Reviewer 2: i like it.

Reviewer: Me too!

<a name="glossary:negotiated-discount"><b>Negotiated Discount</b></a>

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified through an agreement between the customer and the service provider. Multiple Negotiated Discounts may be applied to a single charge at a time, and they may overlap with a Commitment Discount.

Author: ```suggestion
A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified through an agreement between the customer and the service provider. Multiple negotiated discounts may be applied to a single charge at a time, and they may overlap with a [*commitment discount*](#glossary:commitment-discount).
```

<a name="glossary:negotiated-discount"><b>Negotiated Discount</b></a>

> A [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified through an agreement between the customer and the service provider. Multiple negotiated discounts may be applied to a single charge at a time, and they may overlap with a [*commitment discount*](#glossary:commitment-discount).

Reviewer: These two definitions next to each other run very similarly until the second half. Can we add a modifier that immediately differentiates them? For example:

```suggestion
A privately agreed [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified from standard rates. Multiple negotiated discounts may be applied to a single charge at a time, and they may overlap with a [*commitment discount*](#glossary:commitment-discount).
```

Author: see [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1964/files#r2923927510) for reasoning

```suggestion
A negotiated [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified from standard. Multiple negotiated discounts may be applied to a single charge at a time, and they may overlap with a [*commitment discount*](#glossary:commitment-discount).
```

Reviewer: I don't like trying to define a term by immediately using the word in the name.  To say that a _negotiated discount_ is "A negotiated contract commitment..."

I'm happy to wordsmith those first few opening words, but the word "negotiated" specifically doesn't work for me.

Author: @shawnalpay would this work for you?

```suggestion
A privately agreed [*contract commitment*](#glossary:contract-commitment) that provides discounted pricing in exchange for a commitment to specific spend or usage goals over a specified [*period*](#glossary:period), with terms and pricing specifically modified from standard. Multiple negotiated discounts may be applied to a single charge at a time, and they may overlap with a [*commitment discount*](#glossary:commitment-discount).
```

Reviewer: Yep!

A comprehensive list of prices offered by a service provider.

> <a name="glossary:public-commitment-discount-no-carry-over"><b>Public Commitment Discount (No Carry-Over)</b></a>

Reviewer: @ijurica 
Why the `(No Carry-Over)` bit?  It feels tacked on.  We don't have a Contract Commitment property specifying whether a commitment has a carry over, but this implies that we need one.  Do you think so?

Further: I get that `Public Commitment Discount (No Carry-Over)` might feel like the most concise way to describe this concept -- but we are using the term `Commitment Discount` in [a lot of places in the spec](https://github.com/search?q=org%3AFinOps-Open-Cost-and-Usage-Spec%20%22commitment%20discount%22&type=code), not least of which as a prefix in several columns.  I feel fairly uncomfortable with making such a change in the glossary to make it seem like we are trying to sell `Public Commitment Discount (No Carry-Over)` as the term of choice if we are going to continue to use the term `Commitment Discount` as the short-hand term throughout the spec.

I think my preference would be to be more specific with the definition of Commitment Discount in the glossary, but not to try and make changes throughout the spec.  We can discuss more in TF-1 tomorrow. 

@robmartin33 Curious your thoughts here.

Reviewer: I will bring this to @robmartin33 to discuss further.

Reviewer 2: I have changed my opinion on this after review and discussion. I think on the FinOps side we will not be able to redefine Commitment Discount in this way, and I was not intending to propose that change on that side. The proposed term incorporated aspects of the Contract Commitment attributes and so provides a technically clearer differentiation of this in a non-cloud native context, but at the expense of making the term less recognizable. 

I think a middle ground would be to use the descriptors in the proposed name in the definition of Commitment Discount, including a description that says a Cloud Commitment Discount would be a Public Discount with a Continuous delivery that has no carry over, etc. etc. so that the clarity is provided in the definition rather than in the term change.

Reviewer: So then, how about we describe a Commitment Discount as a particular flavor of a Contract Commitment?

A `Commitment Discount` is a `Contract Commitment` with the following designations:

* Typically offered by cloud service providers (e.g., AWS, Azure, GCP, OCI).
* A `Continuous` Contract Commitment Model.
* A `Discount` Contract Commitment Benefit Category.
* A `Public` Contract Commitment Offer Category.
* No benefit carry-over across Contract Commitment Fulfillment Intervals.

If so, then we could continue to carry `Commitment Discount` as a term in our specification without having to change language anywhere else in the spec.

It would be an explicit recognition that FOCUS started out as a specification to represent the billing data for cloud service providers, but that the aperture of FOCUS has now widened to include billing data for all technology providers.

Author: If we define Commitment Discount as a particular flavor of Contract Commitment, I have two clarification questions:

1. Does this imply that Commitment Discounts must now be explicitly represented through the ContractApplied column and the ContractCommitment dataset? Up until now, this has been somewhat implicit. If they are a flavor of Contract Commitment, I assume we are aligning them with the same modeling approach.

2. In cases where the specification currently uses the wording “commitment discount but not negotiated”, would we continue emphasizing that distinction?

3. Do we need to update the Negotiated Discount glossary term as well?

Reviewer: Discussed with @ijurica.  

1. Yes.
2. Leave as-is.
3. Agree that both Commitment Discount and Negotiated Discount should be framed in terms of what type of contract commitment they are (e.g., Offer Category of `Public`, Offer Category of `Negotiated`).

Author: PR updated per this agreement. Resolving this thread
